libiocage/lib/Host.py

```diff
@@ -32,18 +32,16 @@
 
 class HostGenerator:
 
-    _class_distribution = libiocage.lib.Distribution.DistributionGenerator
-
     def __init__(self, root_dataset=None, zfs=None, logger=None):
 
-        libiocage.lib.helpers.init_logger(self, logger)
-        libiocage.lib.helpers.init_zfs(self, zfs)
+        self.logger = libiocage.lib.helpers.init_logger(logger)
+        self.zfs = libiocage.lib.helpers.init_zfs(zfs)
         self.datasets = libiocage.lib.Datasets.Datasets(
             root=root_dataset,
             logger=self.logger,
             zfs=self.zfs
         )
-        self.distribution = self._class_distribution(
+        self.distribution = libiocage.lib.Distribution.DistributionGenerator(
```

Comment:
It is important to instantiate from the class attribute _class_distribution. If you use a non-generator class like Host, you don't expect a generator distribution.

Reply:
@igalic as we discussed offline, this change should be reverted